### PR TITLE
MGP

### PR DESCRIPTION
**Pull request by nknudde**
In this pull request I implemented the Approximatively Marginalised GP as described as indicated in issue #39 . It currently supports multi-output GPs. A notebook and some tests are included.

### CODE REVIEW

**Comment by icouckuy** [comment]
Awesome, thanks. I see that the Travis tests aren't triggered because of an external PR.

In init.py you removed the objective import by accident I think, as well as forgot to remove testmgp.py

But we will fix that once I moved the branch and created a new PR

**Comment by javdrher** [comment]
This is not to be included in the 0.1 release, preferable ater tf 1.3 actual release

I suggest to fix things first, then create an in-house branch as @nknudde has no rights to push after that. Some things to consider:
- we keep MGP in a file models or a file mgp.py? I favor the current setup as it will allow us to do the same thing we applied to acquisition (subpackage) if the file gets big
- This requires tf 1.3 as the Hessian of the Cholesky won't work prior to that. Should we bump the version of GPflowOpt, or just skip this test if executed on a lower version? As i guess we would test on travis on 1.3 it seems natural to actually recommend tf 1.3, mentioning that anything starting from 1.0 should work apart from the mgp but is no longer tested that way.

File: GPflowOpt/models.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        """
+        Compute the mean and variance of held-out data at the points Xnew
+        """
+        theta = self._predict_y_AF_storage['free_vars']
+        pred_f_mean, pred_f_var = self.wrapped.build_predict(Xnew)
+        fmean, fvar = self.wrapped.likelihood.predict_mean_and_var(pred_f_mean, pred_f_var)
+        return self.build_predict(fmean, fvar, theta)
+
+    @AutoFlow((float_type, [None, None]), (float_type, [None, None]))
+    def predict_density(self, Xnew, Ynew):
```
Comment (javdrher) [inline GPflowOpt/models.py]: Still missing a test for predict density :)

**Comment by javdrher** [comment]
Small general guideline: lets avoid rebases. For this PR I'm ok with it, and we won't start from scratch.

Rebases are actually very good but for normal circumstances I think they make the history confusing. its a very good tool for complex merge scenario with an enormous amount of conflicts expected as it allows you to resolve conflicts as they occur, commit by commit (think of a fork which has gone its own way for a while and then is merged). For normal scenarios a merge is preferred.

File: .travis.yml [hunk 2]
```diff
@@ -8,7 +8,7 @@ python:
 cache: pip
 install:
   - pip install -U pip wheel
-  - pip install tensorflow==1.0.1
+  - pip install tensorflow==1.3.0rc0
```
Comment (icouckuy) [inline .travis.yml]: can we later specify this without a version number so we just always test against the latest stable? Which is the goal anyways, or rather work with the latest stable GPflow which hopefully follows the latest stable TF closely.

**Comment by icouckuy** [comment]
I'm also against rebases (at least I am now, some bad experiences). I thought it might be necessary to fix the commit history a bit as i thought it was corrupted due to rebranching, but it actually looks okay ( though there are several commits with the same name)

**Pull request closed without merging**